### PR TITLE
Publish TWIM 20220805

### PR DESCRIPTION
**Pull request by anoadragon453**
Preview: https://pr1397--matrix-org-previews.netlify.app

### CODE REVIEW

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+[madlittlemods (Eric Eastwood)](https://matrix.to/#/@madlittlemods:matrix.org) announces
+
+> ## Matrix public archive
+>
+> This week I'm showing off an early look at the [Matrix public archive](https://github.com/matrix-org/matrix-public-archive). As the name suggests, it acts as an archive of history for your world-readable Matrix rooms. This allows you to view historical content day-by-day and jump back years ago to see what your Matrix room was up to.
+>
+> More importantly, it also allows Google to do the same thing so you’ll probably start finding Matrix content from your favorite search engine and be able to harness the massive knowledge base stored in Matrix. Imagine seeing Matrix logs instead of Stack Overflow answers when googling questions! The new portal into the Matrix ecosystem 🌌
+>
+> You can also see this demoed in [this week's Matrix Live video](https://www.youtube.com/watch?v=6KHQSeJTXm0) (at the top of the page).
```
Comment (MadLittleMods) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:284]: Thank you for replacing this todo link ♥

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+
+### Helm Chart ([website](https://gitlab.com/ananace/charts))
+
+Matrix Kubernetes applications packaged into helm charts
+
+[Ananace](https://matrix.to/#/@ace:kittenface.studio) announces
+
+> ## Kubernetes
+>
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:132]:
```suggestion
```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+> This week has seen the usual updates to [my Helm Charts](https://gitlab.com/ananace/charts) - with element-web being updated to 1.11.2 and matrix-synapse to 1.64.0.
+> Additionally, the matrix-synapse chart now also allows for adding entirely custom .well-known data - along with an example on how to use that for MSC1929.
+
+## Dept of Bridges 🌉
+
+[ChristianP](https://matrix.to/#/@christianp:vector.modular.im) announces
+
+> ## matrix-appservice-discord 2.0.0
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:140]: Worth moving above "ChristianP announces"

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+> Looks like a good contender for one to push over the line given a bit of review and response!
+
+[tulir](https://matrix.to/#/@tulir:maunium.net) reports
+
+> TWIM: someone in [#matrix:matrix.org](https://matrix.to/#/#matrix:matrix.org) complained that the image in the spec section is always the same, so I made this gif (actually webp) out of all those images:
+>
+> <img style="width: 35em" src="/blog/img/qKNVGEePdLFTjTyuDhaUcFud.webp"/>
+
+## Dept of Servers 🏢
+
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:61]:
```suggestion

### hghs ([website](https://github.com/heusalagroup/hghs))

```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+> * 1.4.31 is rolling out which includes the new and improved FTUE onboarding experience along with fixes for markdown lists no longer always starting from 1 and html entities showing up in messages.
+> * The new app layout is starting to materialise with PRs available on github for anyone interested in having a sneak peek!
+> * We’re continuing to make improvements to Live Location sharing and cross signing verification as well as investigating performance issues.
+
+## Dept of Non Chat Clients 🎛️
+
+[ben](https://matrix.to/#/@gnunicorn:matrix.org) reports
+
+> ## [Effektio](https://effektio.org/) — get organized!
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:197]:
```suggestion
```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+Secure and independent communication for Android, connected via Matrix. Come talk with us in [#element-android:matrix.org](https://matrix.to/#/#element-android:matrix.org)!
+
+[adam](https://matrix.to/#/@ouchadam:matrix.org) announces
+
+> * 1.4.31 is rolling out which includes the new and improved FTUE onboarding experience along with fixes for markdown lists no longer always starting from 1 and html entities showing up in messages.
+> * The new app layout is starting to materialise with PRs available on github for anyone interested in having a sneak peek!
+> * We’re continuing to make improvements to Live Location sharing and cross signing verification as well as investigating performance issues.
+
+## Dept of Non Chat Clients 🎛️
+
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:194]:
```suggestion

### Effektio ([website](https://effektio.org))

```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+>
+> 📰 You want updates more frequently and close to when they actually happen? Join our [Effektio Matrix News room](https://matrix.to/#/#news:effektio.org), discuss general aspect in our [foyer](https://matrix.to/#/#foyer:effektio.org) or hang out with the devs in our [tech channel](https://matrix.to/#/#tech:effektio.org).
+
+### Populus Viewer ([website](https://github.com/opentower/populus-viewer))
+
+A Social Annotation Tool Powered by Matrix
+
+[gleachkr](https://matrix.to/#/@gleachkr:matrix.org) reports
+
+> ## Populus-Viewer
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:209]:
```suggestion
```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+>
+> After a short vacation, I've done some new work on populus-viewer. Most of this has been UX and bugfixes, but I've added one new feature that I wanted to share. [MSC3775: Markup Locations for Audiovisual Media](https://github.com/opentower/matrix-doc/blob/audio-markup/proposals/3775-markup-locations-for-audiovisual-media.md) gives a spec for annotating audio and video on matrix, but it also allows you to annotate images. So, for completeness sake, I've added support for annotating image files in Populus! This might be useful for discussing publication layouts, product designs, or for teaching art history. So populus can now annotate: video, audio, images and pdfs.
+>
+> As always, if you want to learn more, follow populus development, or discuss the future of decentralized social annotation on Matrix, come join us at [#opentower:matrix.org](https://matrix.to/#/#opentower:matrix.org).
+> <iframe width="560" height="315" src="https://www.youtube.com/embed/N-l3QYsR_Lo" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+## Dept of Widgets 🧩
+
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:217]:
```suggestion

### matrix-widget-api

```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 9]
```diff
@@ -0,0 +1,324 @@
...
+[Robin](https://matrix.to/#/@robin:robin.town) says
+
+> ## matrix-widget-api
+>
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:221]:
```suggestion
```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+> It also comes with a couple of new features ✨ for widget authors: Sending and receiving to-device messages with [MSC3819](https://github.com/matrix-org/matrix-spec-proposals/pull/3819), and getting TURN servers from the client with [MSC3846](https://github.com/matrix-org/matrix-spec-proposals/pull/3846/). Together, these features enable some new complex use-cases, such as doing VoIP from inside a widget. See Matrix Live for a preview of what that looks like with Element Call! 📹️
+>
+> This is as good a time as any to mention that matrix-widget-api doesn't have to just be for web-based apps. If you're building mobile apps with Matrix and want to make use of widgets, running matrix-widget-api inside a web view to liaison with the actual widget can make it a lot simpler to start supporting widget API features. If you're curious, watch Element iOS+Android for upcoming examples of this.
+
+## Dept of VoIP 🤙
+
+[Dave](https://matrix.to/#/@dave:matrix.org) reports
+
+> ## Element Call
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:232]:
```suggestion
```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 11]
```diff
@@ -0,0 +1,324 @@
...
+>
+> [matrix-widget-api](https://github.com/matrix-org/matrix-widget-api/) v1.0.0 was released yesterday, reflecting the fact that we're not expecting any big changes to the library's architecture for the foreseeable future, and that it's more or less ready for wider use.
+>
+> It also comes with a couple of new features ✨ for widget authors: Sending and receiving to-device messages with [MSC3819](https://github.com/matrix-org/matrix-spec-proposals/pull/3819), and getting TURN servers from the client with [MSC3846](https://github.com/matrix-org/matrix-spec-proposals/pull/3846/). Together, these features enable some new complex use-cases, such as doing VoIP from inside a widget. See Matrix Live for a preview of what that looks like with Element Call! 📹️
+>
+> This is as good a time as any to mention that matrix-widget-api doesn't have to just be for web-based apps. If you're building mobile apps with Matrix and want to make use of widgets, running matrix-widget-api inside a web view to liaison with the actual widget can make it a lot simpler to start supporting widget API features. If you're curious, watch Element iOS+Android for upcoming examples of this.
+
+## Dept of VoIP 🤙
+
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:229]:
```suggestion

### Element Call ([website](https://github.com/vector-im/element-call))

```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 12]
```diff
@@ -0,0 +1,324 @@
...
+
+### matrix-rust-sdk ([website](https://github.com/matrix-org/matrix-rust-sdk))
+
+Matrix Client-Server SDK for Rust
+
+[ben](https://matrix.to/#/@gnunicorn:matrix.org) reports
+
+> ## [Matrix Rust SDK](https://github.com/matrix-org/matrix-rust-sdk)
+>
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:249]:
```suggestion
```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 13]
```diff
@@ -0,0 +1,324 @@
...
+> 👉️ Wanna hack on matrix rust? Go check out our [`help wanted` tagged issues](https://github.com/matrix-org/matrix-rust-sdk/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) and join our matrix channel at Matrix Rust SDK.
+
+### Polyjuice ([website](https://gitlab.com/polyjuice))
+
+Elixir libraries related to the Matrix communications protocol.
+
+[uhoreg](https://matrix.to/#/@hubert:uhoreg.ca) announces
+
+> ## Polyjuice Client Test 🧙
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:262]:
```suggestion
```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 14]
```diff
@@ -0,0 +1,324 @@
...
+> * compatibility with some Matrix clients has been improved
+> * added some functions to make it easy for tests to completely override Matrix endpoints
+>
+> I'm also happy to say that Polyjuice Client Test has helped find bugs in some clients, which have since been fixed.
+
+## Dept of Services 🚀
+
+[madlittlemods (Eric Eastwood)](https://matrix.to/#/@madlittlemods:matrix.org) announces
+
+> ## Matrix public archive
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:278]:
```suggestion
```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 15]
```diff
@@ -0,0 +1,324 @@
...
+> * several more tests have been added
+> * there have been some some front-end improvements
+> * I've also improved the documentation, done some refactoring, and added some helper/utility modules to make things clearer.  My goal is to make it easier for people who don't know Elixir to be able to read and write tests without running away screaming.
+> * compatibility with some Matrix clients has been improved
+> * added some functions to make it easy for tests to completely override Matrix endpoints
+>
+> I'm also happy to say that Polyjuice Client Test has helped find bugs in some clients, which have since been fixed.
+
+## Dept of Services 🚀
+
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:275]:
```suggestion

### Matrix Public Archive ([website](https://github.com/matrix-org/matrix-public-archive))

```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 16]
```diff
@@ -0,0 +1,324 @@
...
+>
+> Under the hood, we use the [MSC3030 `/timestamp_to_event`](https://github.com/matrix-org/matrix-spec-proposals/pull/3030) endpoint to fetch the messages for a given day and then we sever-side render the events with the [Hydrogen SDK](https://github.com/vector-im/hydrogen-web). Re-using Hydrogen gets us pretty and native(to Element) looking UI and keeps the maintenance burden of supporting new event types in Hydrogen.
+>
+> If you want to follow what’s going on and see how it's coming along, you can checkout the project on GitHub, https://github.com/matrix-org/matrix-public-archive
+
+## Dept of Bots 🤖
+
+[moanos [he/him]](https://matrix.to/#/@moanos:hyteck.de) says
+
+> ## Matrix Registration Bot
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:297]:
```suggestion
```

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 17]
```diff
@@ -0,0 +1,324 @@
...
+>
+> ![](/blog/img/zFISGvYfCIhdqunsrPZMhOF.png)
+>
+>
+> Under the hood, we use the [MSC3030 `/timestamp_to_event`](https://github.com/matrix-org/matrix-spec-proposals/pull/3030) endpoint to fetch the messages for a given day and then we sever-side render the events with the [Hydrogen SDK](https://github.com/vector-im/hydrogen-web). Re-using Hydrogen gets us pretty and native(to Element) looking UI and keeps the maintenance burden of supporting new event types in Hydrogen.
+>
+> If you want to follow what’s going on and see how it's coming along, you can checkout the project on GitHub, https://github.com/matrix-org/matrix-public-archive
+
+## Dept of Bots 🤖
+
```
Comment (thibaultamartin) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:294]:
```suggestion

### Matrix Registration Bot ([website](https://github.com/zeratax/matrix-registration))

```

**Review by thibaultamartin — CHANGES_REQUESTED** [review]

File: gatsby/content/blog/2022/08/2022-08-05-twim.mdx [hunk 18]
```diff
@@ -0,0 +1,324 @@
...
+### Matrix Public Archive ([website](https://github.com/matrix-org/matrix-public-archive))
+
+[madlittlemods (Eric Eastwood)](https://matrix.to/#/@madlittlemods:matrix.org) announces
+
+>
+> This week I'm showing off an early look at the [Matrix public archive](https://github.com/matrix-org/matrix-public-archive). As the name suggests, it acts as an archive of history for your world-readable Matrix rooms. This allows you to view historical content day-by-day and jump back years ago to see what your Matrix room was up to.
+>
+> More importantly, it also allows Google to do the same thing so you’ll probably start finding Matrix content from your favorite search engine and be able to harness the massive knowledge base stored in Matrix. Imagine seeing Matrix logs instead of Stack Overflow answers when googling questions! The new portal into the Matrix ecosystem 🌌
+>
+> You can also see this demoed in [this week's Matrix Live video](https://www.youtube.com/watch?v=6KHQSeJTXm0) (at the top of the page).
```
Comment (MadLittleMods) [inline gatsby/content/blog/2022/08/2022-08-05-twim.mdx:283]: Timestamped

```suggestion
> You can also see this demoed in [this week's Matrix Live video](https://youtu.be/6KHQSeJTXm0?t=583) (at the top of the page).
```

**Review by thibaultamartin — APPROVED** [review]

**Pull request merged**